irodsadmin/management/commands/irodsorphans.py

```diff
@@ -25,184 +26,224 @@
+            r'projects/([a-f0-9]{2})/\1[a-f0-9]{6}-([a-f0-9]{4}-){3}[a-f0-9]{12}$',
+            collection.path,
         )
-    ]
-
-
-def get_project_collections(irods_backend):
-    """Return a list of all study collection names."""
-    return [irods_backend.get_path(p) for p in Project.objects.all()]
-
-
-def is_zone(collection):
-    """
-    Check if a given collection matches the format of path to a landing zone
-    collection.
-    """
-    return '/landing_zones/' in collection.path and re.search(
-        r'^\d{8}_\d{6}', collection.name
-    )
-
-
-def is_assay_or_study(collection):
-    """
-    Check if a given collection matches the format of path to a study or assay
-    collection.
-    """
-    return re.match(
-        r'(assay|study)_[a-f0-9]{8}-([a-f0-9]{4}-){3}[a-f0-9]{12}',
-        collection.name,
-    )
-
-
-def is_project(collection):
-    """
-    Check if a given collection matches the format of path to a project
-    collection.
-    """
-    return re.search(
-        r'projects/([a-f0-9]{2})/\1[a-f0-9]{6}-([a-f0-9]{4}-){3}[a-f0-9]{12}$',
-        collection.path,
-    )
-
-
-def get_orphans(irods, irods_backend, expected, assays):
-    """
-    Return a list of orphans in a given irods session that are not in a given
-    list of expected collections.
-    """
-    orphans = []
-    collections = irods.collections.get('/{}/projects'.format(irods.zone))
-
-    for collection in irods_backend.get_colls_recursively(collections):
-        if (
-            is_zone(collection)
-            or is_assay_or_study(collection)
-            or is_project(collection)
-        ):
-            if collection.path not in expected:
-                orphans.append(collection.path)
-
-    for assay in assays:
-        if not assay.get_plugin():
-            continue
-        with irods_backend.get_session() as irods:
-            for collection in irods_backend.get_child_colls(
-                irods, irods_backend.get_path(assay)
-            ):
-                if collection.path not in expected:
-                    orphans.append(collection.path)
-    return orphans
 
+    def _get_orphans(self, irods, expected, assays):
+        """
+        Return a list of orphans in a given irods session that are not in a given
+        list of expected collections.
+        """
+        # Get a sorted list of all project collections
+        all_project_collections = sorted(
+            self.irods_backend.get_colls_recursively(
+                irods.collections.get('/{}/projects'.format(irods.zone))
+            ),
+            key=lambda coll: coll.path,
+        )
+        # Query for all projects of type PROJECT and store in a dict
+        project_dict = {
+            project.full_title: project.sodar_uuid
+            for project in Project.objects.filter(type=PROJECT_TYPE_PROJECT)
+        }
+        # Sort dict by full_title and extract the UUID values
+        project_list = [str(val) for _, val in sorted(project_dict.items())]
```

Comment:
I''m not sure why this line of code is needed, if we don't use `project_dict` for anything else. You can sort the objects in the query and grab just the UUID into a list, even using `values_list('sodar_uuid', flat=True`) to grab simply the UUID. Thus there should not be a need to iterate through the query results twice.